src/bot/field/main.go

```diff
@@ -402,39 +548,21 @@ func createMovementPayload(direction float32, prevX, prevY, x, y, nbMoveData, fp
 
 	return append(payloadSizeBytes, payload...)
 }
-
 func randomSync(bot *botData) {
 	prevX := bot.x
 	prevY := bot.y
-	nbSyncPerMovement := 9
-	if rand.Intn(100) < moveRatio {
+	if bot.isMoving == false && rand.Intn(100) < moveRatio {
+		bot.isMoving = true
 		nextMoveIsInArea := false
 		tryLimit := 20
 		tryCnt := 0
-		for !nextMoveIsInArea {
-			if tryCnt >= tryLimit {
-				return
-			}
+		for !nextMoveIsInArea && tryCnt < tryLimit {
 			r := float64(movementRange)
 			angle := float64(util.RandomInt(1, 360))
 			theta := (angle / 360.0) * 2.0 * math.Pi
 			newX := int(float64(prevX) + r*float64(math.Cos(theta)))
 			newY := int(float64(prevY) + r*float64(math.Sin(theta)))
-			limitXUp := mapSize / 2
-			limitXDown := -mapSize / 2
-			limitYUp := mapSize / 2
-			limitYDown := -mapSize / 2
-			if prevX < 0 {
-				limitXUp = 0
-			} else {
-				limitXDown = 0
-			}
-			if prevY < 0 {
-				limitYUp = 0
-			} else {
-				limitYDown = 0
-			}
-			if !(newX > limitXUp || newX < limitXDown || newY > limitYUp || newY < limitYDown) {
+			if newX >= -mapSize/2 && newX <= mapSize/2 && newY >= -mapSize/2 && newY <= mapSize/2 {
```

Comment:
This is a minor detail, but I thought it would be better to use variables such as `mapHalfSize :=mapSize/2`.